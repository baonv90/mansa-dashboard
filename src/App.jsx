import React from 'react';
import styled from 'styled-components';

import Profile from './templates/Profile';
import Company from './templates/Company';
import AccountInfo from './templates/AccountInfo';

import Block from './layout/Block';

const Dashboard = styled.div`
  display: flex;
  margin: 1.5rem;
  padding: 1rem;
  border-radius: 10px;
  background: var(--white-color);
  height: calc(100vh - 5rem);
`;

const BlockContainer = styled.div`
  display: flex;
  flex-direction: column;
  background: var(--bg-color);
  padding: 1.5rem .5rem;
  border-radius: 10px;
  height: 40vh;
  margin: .5rem .25rem;
`;

const UserContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: 25%;
`;

const AccountContainer = styled.div`
  display: flex;
  width: 75%;
  padding: 1rem 0rem 1rem 1rem;
  flex-direction: column;
`;

const App = () => (
  <Dashboard>
    <UserContainer>
      <BlockContainer>
        <Profile />
      </BlockContainer>
      <BlockContainer>
        <Company />
      </BlockContainer>
    </UserContainer>

    <AccountContainer>
      <Block title='Accounts'>
        <AccountInfo />
      </Block>
    </AccountContainer>
  </Dashboard>
);

export default App;