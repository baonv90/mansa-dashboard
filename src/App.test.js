import renderer from 'react-test-renderer';
import App from './App';

describe('App', () => {
  it('should match snapshot', async () =>  {
    const component = await renderer.create(<App />);
    
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
