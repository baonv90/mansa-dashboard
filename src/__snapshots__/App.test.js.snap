// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="sc-kfYoZR dgOBhr"
>
  <div
    className="sc-bCwfaz eIwwTq"
  >
    <div
      className="sc-fKgJPI eZmOPW"
    />
    <div
      className="sc-fKgJPI eZmOPW"
    />
  </div>
  <div
    className="sc-iwajpm dhxTMH"
  >
    <span
      className="sc-iCoGMd fHfgJB"
    >
      Accounts
    </span>
  </div>
</div>
`;
